src/components/textInput/textInput.tsx

```diff
@@ -62,6 +72,81 @@ export default function TextInput(props: TextInputProps) {
     setMessageText(e.target.value)
   }
 
+  function handleOnFocusToggle() {
+    setIsFocused(!isFocused)
+  }
+
+  function handleToggleSpeechToText() {
+    const microphone = new window.webkitSpeechRecognition()
+
+    microphone.lang = 'en-US'
+
+    if (isRecording) {
+      microphone.stop()
+      setIsEndingRecording(true)
+      setIsRecording(false)
+    } else {
+      microphone.start()
+      setIsRecording(true)
+    }
+
+    microphone.onstart = () => {
+      setIsRecording(true)
+    }
+
+    microphone.onresult = (event: SpeechRecognitionEvent) => {
+      // eslint-disable-next-line no-console
+      console.log('onresult happened')
+      const currentTranscript = event.results[0][0].transcript
+
+      if (messageText.length > 0) {
+        setMessageText(messageText + ' ' + currentTranscript)
+      } else {
+        setMessageText(currentTranscript)
+      }
+
+      setIsRecording(false)
+    }
+
+    microphone.onerror = (event: SpeechRecognitionErrorEvent) => {
+      console.error('Speech recognition error:', event.error)
+      setIsRecording(false)
+    }
+
+    microphone.onend = () => {
+      setIsEndingRecording(false)
+      setIsRecording(false)
+    }
+  }
+
+  const speechToTextButtonAdornment = {
+    endAdornment: (
+      <InputAdornment position="end">
+        {isEndingRecording ? (
+          <CircularProgress size={24} data-cy="spinner" />
+        ) : (
+          <Tooltip title={`${isRecording ? 'Stop' : 'Start'} speech to text`}>
+            <IconButton
+              data-cy="record-button"
+              onClick={handleToggleSpeechToText}
+              size="small"
+            >
+              {isRecording ? (
+                <MicRoundedIcon color="secondary" />
+              ) : (
+                <MicNoneRoundedIcon
+                  sx={{
+                    color: isFocused ? 'primary.main' : 'primary.light',
+                  }}
+                />
+              )}
```

Comment:
Maybe you could extract this part? There are too many ternary operators in the `speechToTextButtonAdornment`